sql Assignment.txt

```diff
@@ -0,0 +1,1134 @@
+| 2008 | Cisneros | SanJose |    300 | 1007 |
++------+----------+---------+--------+------+
+
+24) Count the number of salespeople currently having orders in the orders table.
+
+select count(distinct Snum) as no_of_salespeople from orders;
++-------------------+
+| no_of_salespeople |
++-------------------+
+|                 5 |
++-------------------+
+
+25) Write a query that produces all customers serviced by salespeople with a commission above 12%. Output the customer’s name,
+
+SELECT customers.cname AS "Customer", customers.city AS "City", salespeople.sname AS "Salesman", salespeople.comm FROM 
+customers,salespeople WHERE customers.snum = salespeople.snum AND salespeople.comm > 12;
+
++----------+---------+----------+------+
+| Customer | City    | Salesman | comm |
++----------+---------+----------+------+
+| Liu      | SanJose | Serres   |   13 |
+| Grass    | Berlin  | Serres   |   13 |
+| Cisneros | SanJose | Rifkin   |   15 |
++----------+---------+----------+------+
+
+26) Find salespeople who have multiple customers.
+
+SELECT * FROM salespeople WHERE snum IN (SELECT DISTINCT snum FROM customers cust WHERE EXISTS(SELECT * FROM customers b 
+      WHERE b.snum=cust.snum AND b.cname<>cust.cname));
++------+--------+---------+------+
+| SNUM | SNAME  | CITY    | COMM |
++------+--------+---------+------+
+| 1001 | Peel   | London  |   12 |
+| 1002 | Serres | SanJose |   13 |
++------+--------+---------+------+
+
+27) Find salespeople with customers located in their own cities.
+
+SELECT * FROM salespeople WHERE city = ANY(SELECT city FROM customers);
++------+--------+---------+------+
+| SNUM | SNAME  | CITY    | COMM |
++------+--------+---------+------+
+| 1001 | Peel   | London  |   12 |
+| 1002 | Serres | SanJose |   13 |
+| 1004 | Motika | London  |   11 |
+| 1008 | Fran   | London  |   25 |
++------+--------+---------+------+
+
+28) Find all salespeople whose name starts with ‘P’ and fourth character is ‘I’.
+
+select * from salespeople where sname like 'p__l%';
++------+-------+--------+------+
+| SNUM | SNAME | CITY   | COMM |
++------+-------+--------+------+
+| 1001 | Peel  | London |   12 |
++------+-------+--------+------+
+
+29) Write a query that uses a subquery to obtain all orders for the customer named ‘Cisneros’. Assume you do not know his customer number.
+select * from orders where amt in (select max(amt) from orders where snum in (select snum from salespeople where sname = 'serres'or sname = 'rifkin')group by snum);
++------+---------+------------+------+------+
+| ONUM | AMT     | ODATE      | CNUM | SNUM |
++------+---------+------------+------+------+
+| 3005 | 5160.45 | 10/03/1990 | 2003 | 1002 |
+| 3006 | 1098.16 | 10/03/1990 | 2008 | 1007 |
++------+---------+------------+------+------+
+
+30) Find the largest orders for Serres and Rifkin.
+select * from orders ord where CNUM =(select CNUM from customers where CNAME="Cisneros");+------+---------+------------+------+------+
```

Comment:
I think this is the query for 29th. And the one written in 29th is for 30th one. Please correct it.